lib/js/src/manager/AppConfig.js

```diff
@@ -315,6 +317,57 @@ class AppConfig {
     getMinimumProtocolVersion () {
         return this._minimumProtocolVersion;
     }
+
+    /**
+     * Load manifest file of embedded Web Application
+     * @param {Object} manifest
+     * @return {AppConfig}
+     */
+    loadManifest (manifest) {
+        if (typeof manifest !== 'object') {
+            throw new Error('Wrong parameter, manifest must be object');
+        }
+        const manifestKeys = Object.keys(manifest);
+        const missingKeys = [
+            'appID',
+            'appName',
+            'category',
+            'minProtocolVersion',
+            'minRPCVersion',
+        ].filter(key => !manifestKeys.includes(key));
+        if (missingKeys.length > 0) {
+            throw new Error(`Required parameters are missing in the manifest: ${missingKeys.join(', ')}`);
+        }
+        manifestKeys.forEach((key) => {
+            switch (key) {
+                case 'appID':
+                    this.setAppId(manifest[key]);
+                    break;
+                case 'appName':
+                    this.setAppName(manifest[key]);
+                    break;
+                case 'category':
+                case 'additionalCategories': {
+                    const categories = (Array.isArray(manifest[key]) ? manifest[key] : [manifest[key]])
+                        .map((elem) => {
+                            const key = AppHMIType.keyForValue(elem);
+                            return key ? AppHMIType[key] : '';
+                        })
+                        .concat(this.getAppTypes() || [])
+                        .filter((elem, pos, arr) => elem && (arr.indexOf(elem) === pos));
```

Comment:
I suggest including a comment here which explains that this filter removes duplicate and empty category values

Reply:
Comment added!